corehq/apps/userreports/expressions/specs.py

```diff
@@ -239,3 +240,26 @@ def __call__(self, item, context=None):
         for property_name, expression in self._compiled_properties.items():
             ret[property_name] = expression(item, context)
         return ret
+
+
+class EvalExpressionSpec(JsonObject):
+    type = TypeProperty('evaluator')
+    statement = StringProperty(required=True)
+    context_variables = DictProperty(required=True)
+
+    def configure(self, context_variables):
+        self._context_variables = context_variables
+
+    def __call__(self, item, context=None):
+        var_dict = self.get_variables(item, context)
+        try:
+            return eval_statements(self.statement, var_dict)
+        except (InvalidExpression, SyntaxError):
```

Comment:
it would be nice to be able to catch `SyntaxError`s at save time rather than runtime